Cleanup *_VERSION_MIN_REQUIRED

annevk opened the pull request:
#### 7258436850202f25f922b65cdc892878360873de
<pre>
Cleanup *_VERSION_MIN_REQUIRED
<a href="https://bugs.webkit.org/show_bug.cgi?id=264817">https://bugs.webkit.org/show_bug.cgi?id=264817</a>

Reviewed by Brent Fulgham and Elliott Williams.

Remove code for unsupported macOS and iOS versions.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.cpp:
* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::determineTrackingPreventionStateInternal):
(WebKit::doesParentProcessHaveTrackingPreventionEnabled):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
* Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.mac.sb.in:
* Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.relocatable.mac.sb.in:

Canonical link: <a href="https://commits.webkit.org/270867@main">https://commits.webkit.org/270867@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/88ebdee46ebb4e69f9054e9429bcc2b9455e94b3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26503 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5118 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27750 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/28695 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24239 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/26910 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6919 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2526 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24197 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/26763 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3981 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22763 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3473 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3517 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/23742 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29197 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/23095 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24158 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24143 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/29786 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/25711 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3554 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1752 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27681 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4988 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/33162 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6401 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4029 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/7187 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3879 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple reviewed (approved):
r=me. @pvollan, it would be great to get your review of the sandbox changes. I'm not sure if there is anything we can do to get decent test coverage from EWS.

brentfulgham reviewed (approved):
A really nice cleanup. Thank you for doing this!

@emw-apple The Mac Sandbox changes are fully tested by EWS and our post-commit bots. I do not see any iOS sandbox changes, so I think this is very well-covered by our test infrastructure.

aproskuryakov commented on Source/JavaScriptCore/API/JSValue.h:
```diff
@@ -549,11 +549,7 @@ Create a JSValue from a CGRect.
  */
 @interface JSValue (PropertyAccess)
 
-#if (defined(__MAC_OS_X_VERSION_MIN_REQUIRED) && __MAC_OS_X_VERSION_MIN_REQUIRED < 101500) || (defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && __IPHONE_OS_VERSION_MIN_REQUIRED < 130000)
```
This is API, and people are presumably still able to set deployment target to before 10.15? If so, this change seems incorrect.

annevk replied:
Can you explain how that makes sense? Are we ensuring all our new APIs work with those older macOS versions? And given the code we're removing elsewhere you'd presumably run into build problems too.

aproskuryakov replied:
When a customer installs newest Xcode with newest SDK, they are still able to select an older deployment target, if they want their software to work on older OS versions. These`..._MIN_REQUIRED` macros are part of what changes then.

So this is not about building WebKit, but about building 3rd party software that uses WebKit. The system framework doesn't change on old OSes, so we need to keep our latest SDK compatible with old WebKit.

I don't know if Xcode even has any limit on how far back one can go. Of course there are some natural boundaries (e.g. one cannot use a deployment target that only supported 32-bit Intel, because Xcode has dropped support for 32-bit Intel).

Now, I'm not sure about what exactly is going on here. It is possible that this code is incorrect, and doesn't actually achieve the goal. You can either revert this particular change for now, or investigate deeper.

emw-apple replied:
> Are we ensuring all our new APIs work with those older macOS versions?

I don't think we have any automation which ensures that API changes are backwards-compatible. We have availability annotations to prevent developers from calling methods that may not exist on their deployment target. And we tend to avoid ABI-sensitive declarations in public headers (i.e. no C++, no C structs). But in general we rely on extra reviewer scrutiny on public headers.

> And given the code we're removing elsewhere you'd presumably run into build problems too.

Well, this is the only public header in the PR. So we can safely remove the other code here, because we don't support building _WebKit itself_ for these older OS versions. But we do support 3rd-party developers using a modern SDK to target very old OS versions.

So I think you need to undo the changes to this file, and let the rest of the PR stand.

aproskuryakov commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -896,7 +896,7 @@
 #if !defined(ENABLE_TRACKER_DISPOSITION) \
     && (PLATFORM(IOS_FAMILY) \
-    || (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000))
+    || PLATFORM(MAC))
```
Does this still mean anything, or can this be simplified further? I do not remember where macCatalyst goes.

annevk replied:
That seems correct, given this file is limited to Apple ports.

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -846,7 +846,7 @@
 #endif
 
-#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 120000) || PLATFORM(IOS_FAMILY)
+#if PLATFORM(MAC) || PLATFORM(IOS_FAMILY)
```
Ditto.

annevk replied:
Here that would also enable it on PLATFORM(GTK), no?

aproskuryakov replied:
I think that we normally use `#if PLATFORM(COCOA)`, not `PLATFORM(MAC) || PLATFORM(IOS_FAMILY)` (but again, I do not remember what macCatalyst is included in).

aproskuryakov commented on Source/WTF/wtf/PlatformHave.h:
```diff
@@ -1611,7 +1611,7 @@
 #if !defined(HAVE_UNIFIED_ASC_AUTH_UI) \
-    && ((PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 110000) \
+    && (PLATFORM(MAC) \
     || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 160000) \
```
We don't need iOS 16 any more.

emw-apple reviewed (changes requested):
@aproskuryakov's point about API headers is a good one. Marking r- to make sure we resolve that issue before merging.

annevk committed:
Cleanup *_VERSION_MIN_REQUIRED
https://bugs.webkit.org/show_bug.cgi?id=264817

Reviewed by Brent Fulgham and Elliott Williams.

Remove code for unsupported macOS and iOS versions.

* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/graphics/avfoundation/FormatDescriptionUtilities.cpp:
* Source/WebKit/GPUProcess/mac/com.apple.WebKit.GPUProcess.sb.in:
* Source/WebKit/NetworkProcess/mac/com.apple.WebKit.NetworkProcess.sb.in:
* Source/WebKit/Shared/Cocoa/DefaultWebBrowserChecks.mm:
(WebKit::determineTrackingPreventionStateInternal):
(WebKit::doesParentProcessHaveTrackingPreventionEnabled):
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
* Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.mac.sb.in:
* Source/WebKit/webpushd/mac/com.apple.WebKit.webpushd.relocatable.mac.sb.in:

Canonical link: https://commits.webkit.org/270867@main
